pkg/build/types/parser/deffile_test.go

```diff
@@ -148,3 +235,144 @@ func TestParseDefinitionFileFailure(t *testing.T) {
 		}))
 	}
 }
+
+// Specific tests to cover some corner cases of IsInvalidSectionError()
+func TestIsInvalidSectionErrors(t *testing.T) {
+
+	// Test of IsInvalidSectionError()
+	dummy_keys := []string{"dummy_key1", "dummy_key2"}
+	myValidErr1 := &InvalidSectionError{dummy_keys, errInvalidSection}
+	myValidErr2 := &InvalidSectionError{dummy_keys, errEmptyDefinition}
+	myInvalidErr := errors.New("My dummy error")
+	if IsInvalidSectionError(myValidErr1) == false ||
+		IsInvalidSectionError(myValidErr2) == false ||
+		IsInvalidSectionError(myInvalidErr) == true {
+		t.Fatal("unexpecter return value for IsInvalidSectionError()")
+	}
+
+	// Test of Error()
+	expectedStr1 := "invalid section(s) specified: " + strings.Join(dummy_keys, ", ")
+	expectedStr2 := "Empty definition file: " + strings.Join(dummy_keys, ", ")
+	if myValidErr1.Error() != expectedStr1 || myValidErr2.Error() != expectedStr2 {
+		t.Fatal("unexpecter result from Error()", myValidErr1.Error())
+	}
+}
+
+// Specific tests to cover some corner cases of PopulateDefinition()
+func TestPopulateDefinition(t *testing.T) {
+	//
+	// Some variables used throughout the tests
+	//
+
+	// We use a specific set of section names to reach some corner cases
+	testMap := make(map[string]string)
+	testMap["files"] = "file1 file2"
+	testMap["labels"] = "label1"
+
+	emptyMap := make(map[string]string)
+
+	//
+	// Test with invalid data
+	//
+	invalidData := new(types.Definition)
+	invalidData.Labels = make(map[string]string)
+	populateDefinition(emptyMap, invalidData)
+
+	//
+	// Test with very specific maps
+	//
+
+	// A structure to store results (not really relevant here)
+	myData := new(types.Definition)
+	myData.Labels = make(map[string]string)
+
+	myerr := populateDefinition(testMap, myData)
+	if myerr != nil {
+		t.Fatal("Test failed while testing populateDefinition()")
+	}
+}
+
+// Specific tests to cover some corners cases of doHeader()
+func TestDoHeader(t *testing.T) {
+	invalidHeader := "headerTest"
```

Comment:
maybe loop over `invalidHeader := []string{"headerTest", "headerTest: invalid"}` to cover both error